docs/content/concepts/authentication.md

```diff
@@ -62,6 +62,229 @@ The activation for the individual tenant:
+   }
+
+   location /DEFAULT/controller/ {
+        if ($ssl_client_verify != SUCCESS) {
+           return 403;
+        }
+
+        proxy_pass http://hawkbit.dev.example.com:8080;
+        proxy_set_header Host $http_host;
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        
+        proxy_set_header X-Ssl-Client-Cn $ssl_client_s_dn_cn;
+        proxy_set_header X-Ssl-Issuer-Hash-1 Hawkbit;
+
+        proxy_request_buffering off;
+        client_max_body_size 1000m;
+   }
+
+   # For clients that is using UI or Management API
+   location / {
+        proxy_pass http://hawkbit.dev.example.com:8080;
+        proxy_set_header Host $http_host; 
+        proxy_set_header X-Real-IP $remote_addr;
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+	    proxy_request_buffering off;
+	    client_max_body_size 1000m;
+   }   
+}
+```
+To authenticate the request to Hawkbit itself, the common name and issuer hash of the presented client certificate are required. The issuer hash of a certificate is the hash of the certificate that signed the client certificate, which in our case is the CA.
+
+You can use the following command to get the issuer hash:
+
+`openssl x509 -in client_certificate.crt -issuer_hash -noout`
+
+However, in the Nginx configuration, obtaining the issuer hash is not possible without addons. Therefore, this header is manually entered as Hawkbit.
+
+When deploying Nginx, you will need a `.yml` file. Here's an example `docker-compose.yml` file for Nginx Docker.
+
+```yml
+version: '3'
+
+services:
+  webserver:
+    image: nginx:latest
+    ports:
+      - 80:80
+      - 443:443
+    restart: always
+    volumes:
+      - ./nginx/conf/:/etc/nginx/conf.d/:ro
+      - ./certbot/www:/var/www/certbot/:ro
+      - ./certbot/conf/:/etc/nginx/ssl/:ro
+      - ./client-cer/:/etc/nginx/client-cer/
+      - ./landing-page/:/etc/webserver/landing-page
+  certbot:
+    image: certbot/certbot:latest
+    volumes:
+      - ./certbot/www/:/var/www/certbot/:rw
+      - ./certbot/conf/:/etc/letsencrypt/:rw
+```
+`/client-cer/:/etc/nginx/client-cer/` is the designated location for the certificate authority that has signed the client certificate. The presented client certificate will be verified against this CA.
+
+#### Swupdate Suricatta Configurations
+
+If the client is utilizing the SWUpdate Suricatta service, the configurations on the device or client side should also be adjusted as follows:
+
+The location of the config file is `/etc/swupdate/swupdate.conf`
+
+```yml
+suricatta :
+{
+tenant = "default";
+id = "[ID]";
+url = "[URL]";
+nocheckcert = false;
+cafile = "[CAFile]";
+sslkey = "/etc/ssl/certs/[ID].key";
+sslcert = "/etc/ssl/certs/[ID].crt";
+};
+```
+
+If your client service is a linux, you can use the command bellow to see the logs produced by the swupdate.
+
+`journalctl --follow -u swupdate`
+
+#### Hawkbit Configurations
```

Comment:
When I tried to upgrade the devices using Mutual TLS without adding these configurations for the URLs because by default, they are generated as follows:

`http://hawkbit.dev.example.com:8080//.......`

The device was attempting to reach the service behind the reverse proxy using HTTP and the port, thus resulting in unauthorized access. With the current configurations, the link is generated as follows, allowing the device to successfully download the firmware:

`https://hawkbit.dev.example.com//.......`

Even the authentication works when adding the devices; without adding these configurations, it will not work when the device is downloading the firmware.

That is why I added these configs. What do you think?

By the way, I removed the docker-compose.yml config file and replaced it with normal config parameters in case someone prefers to add them to the properties file.

Reply:
@avgustinmm Could you please check this comment & and PR again when you time. I applied the neccessary fixes. Thanks.